wp-job-manager-functions.php

```diff
@@ -1662,12 +1662,5 @@ function job_manager_user_can_view_job_listing( $job_id ) {
 		$can_view = true;
 	}
 
-	$key = get_post_meta( $job_id, 'share_link_key', true );
```

Comment:
I checked that this meta exists in the Resumes plugin. Maybe it had some purpose related to the resumes? Or it really shouldn't be here? If it had some purpose, probably we'd need a filter and expand it through Resumes. 🤔

Reply:
I haven't looked closely, but I'm guessing it's related to a way to share Resumes that aren't publicly visible by sending a URL that has a key in it. If we want that for WPJM I think we would need a separate PR.

This is a guess, though.

Reply:
Yeah, reading the Resumes plugin it look related to only resumes, not job advertisements. 

So it's kinda out of scope for the original feature, IMHO.

Reply:
Yes when a resume is used for applications a special link is sent to the listing owner using that share link key so they can view it regardless of the visibility settings.   May not be a bad idea to keep it integrated though, in case a theme or addon plugin is using it in some way.

Reply:
Removing this could have adverse side effects we're not aware of and i see no real benefit of removing it with the potential downsides of doing so

Reply:
Hey there @tripflex, thanks for your comments. 

Some points:

- Testing here, the WP Job Manager Resumes still works successfully (I have tested it locally), even after having this change removed, so it shouldn't break that functionality.
- Sorry, but this phrase:  
> May not be a bad idea to keep it integrated though, in case a theme or addon plugin is using it in some way.

Doesn't "matches" well with this one: 

> Removing this could have adverse side effects we're not aware

Given that having this post meta being checked here is, by definition, a side-effect, given that we're not using the `share_link_key` anywhere else in the plugin nor allowing the user to change it here.

- With that said, if some theme or plugin wants to add that feature anyway, it can always use filter `job_manager_user_can_view_job`, without any issues. 
- And...finally, as mentioned previously, if we want to add that specific feature for WPJM, it's no problem, we just need to add this a separate PR instead, because it will need some additional changes besides having that checked here. 

In other words, I still prefer to keep this out of this feature, as it makes more sense than just having this snippet here, unused. If we want to add that feature (in a more complete way, with changes to the UI and so on) back in the future, we can do that, but in the meantime, I don't think it's worth keeping it here given that we're not using this post meta anywhere else in this plugin.